ilxutils/ilxutils/cli.py

```diff
@@ -0,0 +1,395 @@
+                    score.append(mock_rank.index(pref))
+                else:
+                    score.append(-1)
+            else:
+                score.append(-1)
+
+        new_pref_index = score.index(max(score))
+        new_pref_iri = data['existing_ids'][new_pref_index]['iri']
+        if new_pref_iri.rsplit('/', 1)[0] == 'http://uri.interlex.org/base':
+            if old_pref_index:
+                if old_pref_index != new_pref_index:
+                    return data
+        for e in data['existing_ids']:
+            e['preferred'] = 0
+        data['existing_ids'][new_pref_index]['preferred'] = 1
+        return data
+
+    def add_entity(self, rdf_type, superclass, label, definition=None):
+
+        # Checks if you inputed the right type
+        accepted_types = ['term', 'cde', 'annotation', 'relationship', 'fde']
+        if rdf_type not in accepted_types:
+            error = 'rdf_type must be one of the following: {accepted_types}'
+            return self.test_check(error.format(accepted_types=accepted_types))
+
+        # Pulls superclass data out and checks if it exists
+        superclass_data, success = self.get_data_from_ilx(ilx_id=superclass)
+        if not success:
+            error = '{superclass} is does not exist and cannot be used as a superclass.'
+            return self.test_check(error.format(superclass=superclass))
+
+        # Searchs database to see if the term exists. Will return anything similar,
+        # but we want only what is_equal
+        search_results = self.search_by_label(label)['data']
+        search_results = [sr for sr in search_results
+                          if self.is_equal(sr['label'], label_bug_fix(label))]
+
+        # If search_results is not empty, we need to see if the type and superclass are also a
+        # match. If not, you can create this entity. HOWEVER. If you are the creator of an entity,
+        # you can only have one label of any type or superclass
+        if search_results:
+            entity = search_results[0] # garunteed to only have one match if any
+            entity, success = self.get_data_from_ilx(ilx_id=entity['ilx']) # get all the metadata
+            entity = entity['data']
+            user_url = 'https://scicrunch.org/api/1/user/info?key={api_key}'
+            user_data = self.get(user_url.format(api_key=self.APIKEY))
+            user_data = user_data['data']
+            if str(entity['uid']) == str(user_data['id']): # creator check
+                bp = 'Entity {label} already exisits with ILX ID {ilx_id} and of type {rdf_type}'
+                return self.test_check(bp.format(label=label,
+                                                 ilx_id=entity['ilx'],
+                                                 rdf_type=entity['type']))
+            types_equal = self.is_equal(entity['type'], rdf_type) # type check
+            entity_super_ilx = entity['superclass']['ilx'] if entity['superclass'] else ''
```

Comment:
`interlex post entity term ILX:0106552 example1` breaks this line because ILX:0106552 has no superclasses. Change to `entity_super_ilx = entity['superclass']['ilx'] if 'superclass' in entity and entity['superclass'] else ''`